<!DOCTYPE html><html lang="en"><head><meta charSet="utf-8"/><meta http-equiv="X-UA-Compatible" content="IE=edge"/><title>The Command Line Interface (CLI) · GenoML</title><meta name="viewport" content="width=device-width"/><meta name="generator" content="Docusaurus"/><meta name="description" content="&lt;h2&gt;&lt;a class=&quot;anchor&quot; aria-hidden=&quot;true&quot; id=&quot;at-a-glance&quot;&gt;&lt;/a&gt;&lt;a href=&quot;#at-a-glance&quot; aria-hidden=&quot;true&quot; class=&quot;hash-link&quot;&gt;&lt;svg class=&quot;hash-link-icon&quot; aria-hidden=&quot;true&quot; height=&quot;16&quot; version=&quot;1.1&quot; viewBox=&quot;0 0 16 16&quot; width=&quot;16&quot;&gt;&lt;path fill-rule=&quot;evenodd&quot; d=&quot;M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z&quot;&gt;&lt;/path&gt;&lt;/svg&gt;&lt;/a&gt;At a Glance&lt;/h2&gt;
"/><meta name="docsearch:language" content="en"/><meta property="og:title" content="The Command Line Interface (CLI) · GenoML"/><meta property="og:type" content="website"/><meta property="og:url" content="https://genoml.github.io/index.html"/><meta property="og:description" content="&lt;h2&gt;&lt;a class=&quot;anchor&quot; aria-hidden=&quot;true&quot; id=&quot;at-a-glance&quot;&gt;&lt;/a&gt;&lt;a href=&quot;#at-a-glance&quot; aria-hidden=&quot;true&quot; class=&quot;hash-link&quot;&gt;&lt;svg class=&quot;hash-link-icon&quot; aria-hidden=&quot;true&quot; height=&quot;16&quot; version=&quot;1.1&quot; viewBox=&quot;0 0 16 16&quot; width=&quot;16&quot;&gt;&lt;path fill-rule=&quot;evenodd&quot; d=&quot;M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z&quot;&gt;&lt;/path&gt;&lt;/svg&gt;&lt;/a&gt;At a Glance&lt;/h2&gt;
"/><meta property="og:image" content="https://genoml.github.io/img/docusaurus.png"/><meta name="twitter:card" content="summary"/><meta name="twitter:image" content="https://genoml.github.io/img/docusaurus.png"/><link rel="shortcut icon" href="/img/favicon.png"/><link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css"/><script>
              (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
              (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
              m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
              })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

              ga('create', 'UA-134657372-1', 'auto');
              ga('send', 'pageview');
            </script><script type="text/javascript" src="https://buttons.github.io/buttons.js"></script><link rel="stylesheet" href="/css/main.css"/><script src="/js/codetabs.js"></script></head><body class="sideNavVisible separateOnPageNav"><div class="fixedHeaderContainer"><div class="headerWrapper wrapper"><header><a href="/"><img class="logo" src="/img/dna.png" alt="GenoML"/><h2 class="headerTitleWithLogo">GenoML</h2></a><div class="navigationWrapper navigationSlider"><nav class="slidingNav"><ul class="nav-site nav-site-internal"><li class="siteNavGroupActive"><a href="/docs/introduction" target="_self">About</a></li><li class="siteNavGroupActive"><a href="/docs/quick-start" target="_self">Getting Started</a></li><li class="siteNavGroupActive"><a href="/docs/contribute" target="_self">Contribute</a></li><li class=""><a href="https://github.com/GenoML" target="_self">GitHub</a></li></ul></nav></div></header></div></div><div class="navPusher"><div class="docMainWrapper wrapper"><div class="container docsNavContainer" id="docsNav"><nav class="toc"><div class="toggleNav"><section class="navWrapper wrapper"><div class="navBreadcrumb wrapper"><div class="navToggle" id="navToggler"><i></i></div><h2><i>›</i><span>API Reference</span></h2><div class="tocToggler" id="tocToggler"><i class="icon-toc"></i></div></div><div class="navGroups"><div class="navGroup"><h3 class="navGroupCategoryTitle">GenoML</h3><ul class=""><li class="navListItem"><a class="navItem" href="/docs/introduction">Introduction</a></li><li class="navListItem"><a class="navItem" href="/docs/principles">Principles and Philosophy</a></li><li class="navListItem"><a class="navItem" href="/docs/roadmap">Roadmap</a></li><li class="navListItem"><a class="navItem" href="/docs/contribute">Contributing</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">Getting Started</h3><ul class=""><li class="navListItem"><a class="navItem" href="/docs/quick-start">Quick start</a></li><li class="navListItem"><a class="navItem" href="/docs/usage">Usage</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">API Reference</h3><ul class=""><li class="navListItem navListItemActive"><a class="navItem" href="/docs/cli">The Command Line Interface (CLI)</a></li></ul></div></div></section></div><script>
            var coll = document.getElementsByClassName('collapsible');
            var checkActiveCategory = true;
            for (var i = 0; i < coll.length; i++) {
              var links = coll[i].nextElementSibling.getElementsByTagName('*');
              if (checkActiveCategory){
                for (var j = 0; j < links.length; j++) {
                  if (links[j].classList.contains('navListItemActive')){
                    coll[i].nextElementSibling.classList.toggle('hide');
                    coll[i].childNodes[1].classList.toggle('rotate');
                    checkActiveCategory = false;
                    break;
                  }
                }
              }

              coll[i].addEventListener('click', function() {
                var arrow = this.childNodes[1];
                arrow.classList.toggle('rotate');
                var content = this.nextElementSibling;
                content.classList.toggle('hide');
              });
            }

            document.addEventListener('DOMContentLoaded', function() {
              createToggler('#navToggler', '#docsNav', 'docsSliderActive');
              createToggler('#tocToggler', 'body', 'tocActive');

              const headings = document.querySelector('.toc-headings');
              headings && headings.addEventListener('click', function(event) {
                if (event.target.tagName === 'A') {
                  document.body.classList.remove('tocActive');
                }
              }, false);

              function createToggler(togglerSelector, targetSelector, className) {
                var toggler = document.querySelector(togglerSelector);
                var target = document.querySelector(targetSelector);

                if (!toggler) {
                  return;
                }

                toggler.onclick = function(event) {
                  event.preventDefault();

                  target.classList.toggle(className);
                };
              }
            });
        </script></nav></div><div class="container mainContainer"><div class="wrapper"><div class="post"><header class="postHeader"><h1 class="postHeaderTitle">The Command Line Interface (CLI)</h1></header><article><div><span><h2><a class="anchor" aria-hidden="true" id="at-a-glance"></a><a href="#at-a-glance" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>At a Glance</h2>
<p><code>genoml</code> lets them know you are running a workflow supported by GenoML and its growing community.</p>
<p>Then let us know if you would like to work with a continuous or discrete outcome in your *.pheno file with <code>genoml continuous</code> or <code>genoml discrete</code>.</p>
<p>The current iteration of documentation supports the <code>supervised</code> learning workflow, the list of workflows incoming is scary!
After that you can add one of the subcommands including choice of <code>munge</code>, <code>train</code>, <code>tune</code>, or <code>test</code>, these are steps within the supervised workflow.</p>
<p>If you are interested in harmonizing a test dataset for external validation of a previous model, use <code>genoml harmonize</code>.</p>
<p>The general structure of GenoML commands is <code>genoml</code> then <code>outcome</code> type (discrete or continuous for now), then <code>workflow</code> (supervised for now), then <code>subcommand</code> that is usually a step within a larger workflow.</p>
<p>Detailed command line options relating to published GenoML <code>workflows</code> and their <code>subcommands</code> below.</p>
<h2><a class="anchor" aria-hidden="true" id="genoml-the-root-of-all-commands"></a><a href="#genoml-the-root-of-all-commands" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>genoml - the root of all commands</h2>
<pre><code class="hljs">usage: genoml &lt;command&gt; [&lt;args&gt;]
   continuous      <span class="hljs-keyword">for</span> processing continuous datatypes (ex: age at onset)
   discrete        <span class="hljs-keyword">for</span> processing discrete datatypes (ex: case vs. control status)
   harmonize       <span class="hljs-keyword">for</span> harmonizing<span class="hljs-built_in"> incoming </span>test datasets <span class="hljs-keyword">to</span> use the same SNPs <span class="hljs-keyword">and</span> reference alleles prior <span class="hljs-keyword">to</span> munging, training, <span class="hljs-keyword">and</span> testing

genoml

positional arguments:
  command     Subcommand <span class="hljs-keyword">to</span> <span class="hljs-builtin-name">run</span>
</code></pre>
<h2><a class="anchor" aria-hidden="true" id="genoml-harmonize"></a><a href="#genoml-harmonize" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>genoml harmonize</h2>
<pre><code class="hljs">usage: genoml harmonize [-h] --test_geno_prefix TEST_GENO_PREFIX
                        [--test_prefix TEST_PREFIX]
                        [--ref_model_prefix REF_MODEL_PREFIX]
                        --training_snps_alleles TRAINING_SNPS_ALLELES [-v]

optional arguments:
  -h, --help            show this help message <span class="hljs-keyword">and</span> exit
  --test_geno_prefix TEST_GENO_PREFIX
                       <span class="hljs-built_in"> Prefix </span>of the genotypes <span class="hljs-keyword">for</span> the test dataset <span class="hljs-keyword">in</span> PLINK
                        binary format.
  --test_prefix TEST_PREFIX
                       <span class="hljs-built_in"> Prefix </span><span class="hljs-keyword">for</span> the dataset you would like <span class="hljs-keyword">to</span> test against
                        your reference model. Remember, the model will <span class="hljs-keyword">not</span>
                        function well <span class="hljs-keyword">if</span> it does <span class="hljs-keyword">not</span> include the same
                        features, <span class="hljs-keyword">and</span> these features should be on the same
                        numeric scale, you can leave off the <span class="hljs-string">'.dataForML.h5'</span>
                        suffix.
  --ref_model_prefix REF_MODEL_PREFIX
                       <span class="hljs-built_in"> Prefix </span>of your reference model file, you can leave off
                        the <span class="hljs-string">'.joblib'</span> suffix.
  --training_snps_alleles TRAINING_SNPS_ALLELES
                        File <span class="hljs-keyword">to</span> the SNPs <span class="hljs-keyword">and</span> alleles file generated <span class="hljs-keyword">in</span> the
                        training phase that we will use <span class="hljs-keyword">to</span> compare.
  -v, --verbose         Verbose output.
</code></pre>
<h2><a class="anchor" aria-hidden="true" id="genoml-discrete-supervised-munge"></a><a href="#genoml-discrete-supervised-munge" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>genoml discrete supervised munge</h2>
<pre><code class="hljs">usage: genoml discrete supervised munge [-h] [--prefix PREFIX]
                                        [--impute {median,mean}] [--geno GENO]
                                        --pheno PHENO [--addit ADDIT]
                                        [--feature_selection FEATURE_SELECTION]
                                        [--gwas GWAS] [--p P] [--vif VIF]
                                        [--iter ITER]
                                        [--ref_cols_harmonize REF_COLS_HARMONIZE]
                                        [-v]

optional arguments:
  -h, --help            show this help message <span class="hljs-keyword">and</span> exit
  --prefix<span class="hljs-built_in"> PREFIX </span>     <span class="hljs-built_in"> Prefix </span><span class="hljs-keyword">for</span> your output build.
  --impute {median,mean}
                        Imputation: (mean, median). Governs secondary
                        imputation <span class="hljs-keyword">and</span> data transformation [default: median].
  --geno GENO           Genotype: (string file path). Path <span class="hljs-keyword">to</span> PLINK format
                        genotype file, everything before the *.bed/bim/fam
                        [default: None].
  --pheno PHENO         Phenotype: (string file path). Path <span class="hljs-keyword">to</span> CSV phenotype
                        file [default: lost].
  --addit ADDIT         Additional: (string file path). Path <span class="hljs-keyword">to</span> CSV format
                        feature file [default: None].
  --feature_selection FEATURE_SELECTION
                        <span class="hljs-builtin-name">Run</span> a quick tree-based feature selection routine prior
                        <span class="hljs-keyword">to</span> anything <span class="hljs-keyword">else</span>, here you input the integer number of
                        estimators needed, we suggest &gt;= 50. The<span class="hljs-built_in"> default </span>of 0
                        will skip this functionality. This will also output a
                        reduced dataset <span class="hljs-keyword">for</span> analyses <span class="hljs-keyword">in</span> addition <span class="hljs-keyword">to</span> feature
                        ranks. [default: 0]
  --gwas GWAS           GWAS summary stats: (string file path). Path <span class="hljs-keyword">to</span> CSV
                        format external GWAS summary statistics containing at
                        least the columns SNP <span class="hljs-keyword">and</span> P <span class="hljs-keyword">in</span> the header [default:
                        nope].
  --p P                 P threshold <span class="hljs-keyword">for</span> GWAS: (some value between 0-1). P
                        value <span class="hljs-keyword">to</span><span class="hljs-built_in"> filter </span>your SNP data on [default: 0.001].
  --vif VIF             Variance Inflation Factor (VIF): (integer). This is
                        the VIF threshold <span class="hljs-keyword">for</span> pruning non-genotype features.
                        We recommend a value of 5-10. The<span class="hljs-built_in"> default </span>of 0 means
                        <span class="hljs-literal">no</span> VIF filtering will be done. [default: 0].
  --iter ITER           Iterator: (integer). How many iterations of VIF
                        pruning of features <span class="hljs-keyword">do</span> you want <span class="hljs-keyword">to</span> run. <span class="hljs-keyword">To</span> save time
                        VIF is <span class="hljs-builtin-name">run</span> <span class="hljs-keyword">in</span> randomly assorted chunks of 1000
                        features per iteration. The<span class="hljs-built_in"> default </span>of 1 means only
                        one pass through the data. [default: 1].
  --ref_cols_harmonize REF_COLS_HARMONIZE
                        Are you now munging a test dataset following the
                        harmonize <span class="hljs-keyword">step</span>? Here you input the path <span class="hljs-keyword">to</span> the <span class="hljs-keyword">to</span> the
                        *_refColsHarmonize_toKeep.txt file generated at that
                        <span class="hljs-keyword">step</span>.
  -v, --verbose         Verbose output.
</code></pre>
<h2><a class="anchor" aria-hidden="true" id="genoml-continuous-supervised-munge"></a><a href="#genoml-continuous-supervised-munge" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>genoml continuous supervised munge</h2>
<pre><code class="hljs">usage: genoml continuous supervised munge [-h] [--prefix PREFIX]
                                          [--impute {median,mean}]
                                          [--geno GENO] --pheno PHENO
                                          [--addit ADDIT]
                                          [--feature_selection FEATURE_SELECTION]
                                          [--gwas GWAS] [--p P] [--vif VIF]
                                          [--iter ITER]
                                          [--ref_cols_harmonize REF_COLS_HARMONIZE]
                                          [-v]

optional arguments:
  -h, --help            show this help message <span class="hljs-keyword">and</span> exit
  --prefix<span class="hljs-built_in"> PREFIX </span>     <span class="hljs-built_in"> Prefix </span><span class="hljs-keyword">for</span> your output build.
  --impute {median,mean}
                        Imputation: (mean, median). Governs secondary
                        imputation <span class="hljs-keyword">and</span> data transformation [default: median].
  --geno GENO           Genotype: (string file path). Path <span class="hljs-keyword">to</span> PLINK format
                        genotype file, everything before the *.bed/bim/fam
                        [default: None].
  --pheno PHENO         Phenotype: (string file path). Path <span class="hljs-keyword">to</span> CSV phenotype
                        file [default: lost].
  --addit ADDIT         Additional: (string file path). Path <span class="hljs-keyword">to</span> CSV format
                        feature file [default: None].
  --feature_selection FEATURE_SELECTION
                        <span class="hljs-builtin-name">Run</span> a quick tree-based feature selection routine prior
                        <span class="hljs-keyword">to</span> anything <span class="hljs-keyword">else</span>, here you input the integer number of
                        estimators needed, we suggest &gt;= 50. The<span class="hljs-built_in"> default </span>of 0
                        will skip this functionality. This will also output a
                        reduced dataset <span class="hljs-keyword">for</span> analyses <span class="hljs-keyword">in</span> addition <span class="hljs-keyword">to</span> feature
                        ranks. [default: 0]
  --gwas GWAS           GWAS summary stats: (string file path). Path <span class="hljs-keyword">to</span> CSV
                        format external GWAS summary statistics containing at
                        least the columns SNP <span class="hljs-keyword">and</span> P <span class="hljs-keyword">in</span> the header [default:
                        nope].
  --p P                 P threshold <span class="hljs-keyword">for</span> GWAS: (some value between 0-1). P
                        value <span class="hljs-keyword">to</span><span class="hljs-built_in"> filter </span>your SNP data on [default: 0.001].
  --vif VIF             Variance Inflation Factor (VIF): (integer). This is
                        the VIF threshold <span class="hljs-keyword">for</span> pruning non-genotype features.
                        We recommend a value of 5-10. The<span class="hljs-built_in"> default </span>of 0 means
                        <span class="hljs-literal">no</span> VIF filtering will be done. [default: 0].
  --iter ITER           Iterator: (integer). How many iterations of VIF
                        pruning of features <span class="hljs-keyword">do</span> you want <span class="hljs-keyword">to</span> run. <span class="hljs-keyword">To</span> save time
                        VIF is <span class="hljs-builtin-name">run</span> <span class="hljs-keyword">in</span> randomly assorted chunks of 1000
                        features per iteration. The<span class="hljs-built_in"> default </span>of 1 means only
                        one pass through the data. [default: 1].
  --ref_cols_harmonize REF_COLS_HARMONIZE
                        Are you now munging a test dataset following the
                        harmonize <span class="hljs-keyword">step</span>? Here you input the path <span class="hljs-keyword">to</span> the <span class="hljs-keyword">to</span> the
                        *_refColsHarmonize_toKeep.txt file generated at that
                        <span class="hljs-keyword">step</span>.
  -v, --verbose         Verbose output.
</code></pre>
<h2><a class="anchor" aria-hidden="true" id="genoml-discrete-supervised-train"></a><a href="#genoml-discrete-supervised-train" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>genoml discrete supervised train</h2>
<pre><code class="hljs">usage: genoml discrete supervised train [-h] [--prefix PREFIX]
                                        [--metric_max {AUC,Balanced_Accuracy,Specificity,Sensitivity}]
                                        [--prob_hist PROB_HIST] [--auc AUC]
                                        [--matching_columns MATCHING_COLUMNS]
                                        [-v]

optional arguments:
  -h, --help            show this help message <span class="hljs-keyword">and</span> exit
  --prefix<span class="hljs-built_in"> PREFIX </span>     <span class="hljs-built_in"> Prefix </span><span class="hljs-keyword">for</span> your output build.
  --metric_max {AUC,Balanced_Accuracy,Specificity,Sensitivity}
                        How <span class="hljs-keyword">do</span> you want <span class="hljs-keyword">to</span> determine which algorithm performed
                        the best? [default: AUC].
  --prob_hist PROB_HIST
  --auc AUC
  --matching_columns MATCHING_COLUMNS
                        This is the list of intersecting columns between
                        reference <span class="hljs-keyword">and</span> testing datasets with the suffix
                        *_finalHarmonizedCols_toKeep.txt
  -v, --verbose         Verbose output.
</code></pre>
<h2><a class="anchor" aria-hidden="true" id="genoml-continuous-supervised-train"></a><a href="#genoml-continuous-supervised-train" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>genoml continuous supervised train</h2>
<pre><code class="hljs">usage: genoml continuous supervised train [-h] [--prefix PREFIX]
                                          [--export_predictions EXPORT_PREDICTIONS]
                                          [--matching_columns MATCHING_COLUMNS]
                                          [-v]

optional arguments:
  -h, --help            show this help message <span class="hljs-keyword">and</span> exit
  --prefix<span class="hljs-built_in"> PREFIX </span>     <span class="hljs-built_in"> Prefix </span><span class="hljs-keyword">for</span> your output build.
  --export_predictions EXPORT_PREDICTIONS
  --matching_columns MATCHING_COLUMNS
                        This is the list of intersecting columns between
                        reference <span class="hljs-keyword">and</span> testing datasets with the suffix
                        *_finalHarmonizedCols_toKeep.txt
  -v, --verbose         Verbose output.
</code></pre>
<h2><a class="anchor" aria-hidden="true" id="genoml-discrete-supervised-tune"></a><a href="#genoml-discrete-supervised-tune" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>genoml discrete supervised tune</h2>
<pre><code class="hljs">usage: genoml discrete supervised tune [-h] [--prefix PREFIX]
                                       [--metric_tune {AUC,Balanced_Accuracy}]
                                       [--max_tune MAX_TUNE] [--n_cv N_CV]
                                       [-v]

optional arguments:
  -h, --help            show this help message <span class="hljs-keyword">and</span> exit
  --prefix<span class="hljs-built_in"> PREFIX </span>     <span class="hljs-built_in"> Prefix </span><span class="hljs-keyword">for</span> your output build.
  --metric_tune {AUC,Balanced_Accuracy}
                        Using what metric of the best algorithm <span class="hljs-keyword">do</span> you want <span class="hljs-keyword">to</span>
                        tune on? [default: AUC].
  --max_tune MAX_TUNE   Max number of tuning iterations: (integer likely
                        greater than 10). This governs the length of tuning
                        process, <span class="hljs-builtin-name">run</span> speed <span class="hljs-keyword">and</span> the maximum number of possible
                        combinations of tuning parameters [default: 50].
  --n_cv N_CV           Number of cross validations: (integer likely greater
                        than 3). Here we <span class="hljs-builtin-name">set</span> the number of cross-validation
                        runs <span class="hljs-keyword">for</span> the algorithms [default: 5].
  -v, --verbose         Verbose output.
</code></pre>
<h2><a class="anchor" aria-hidden="true" id="genoml-continuous-supervised-tune"></a><a href="#genoml-continuous-supervised-tune" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>genoml continuous supervised tune</h2>
<pre><code class="hljs">usage: genoml continuous supervised tune [-h] [--prefix PREFIX]
                                         [--max_tune MAX_TUNE] [--n_cv N_CV]
                                         [-v]

optional arguments:
  -h, --help           show this help message <span class="hljs-keyword">and</span> exit
  --prefix<span class="hljs-built_in"> PREFIX </span>    <span class="hljs-built_in"> Prefix </span><span class="hljs-keyword">for</span> your output build.
  --max_tune MAX_TUNE  Max number of tuning iterations: (integer likely
                       greater than 10). This governs the length of tuning
                       process, <span class="hljs-builtin-name">run</span> speed <span class="hljs-keyword">and</span> the maximum number of possible
                       combinations of tuning parameters [default: 50].
  --n_cv N_CV          Number of cross validations: (integer likely greater
                       than 3). Here we <span class="hljs-builtin-name">set</span> the number of cross-validation
                       runs <span class="hljs-keyword">for</span> the algorithms [default: 5].
  -v, --verbose        Verbose output.
</code></pre>
<h2><a class="anchor" aria-hidden="true" id="genoml-discrete-supervised-test"></a><a href="#genoml-discrete-supervised-test" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>genoml discrete supervised test</h2>
<pre><code class="hljs">usage: genoml discrete supervised test [-h] [--prefix PREFIX]
                                       [--test_prefix TEST_PREFIX]
                                       [--ref_model_prefix REF_MODEL_PREFIX]
                                       [-v]

optional arguments:
  -h, --help            show this help message <span class="hljs-keyword">and</span> exit
  --prefix<span class="hljs-built_in"> PREFIX </span>     <span class="hljs-built_in"> Prefix </span><span class="hljs-keyword">for</span> your output build.
  --test_prefix TEST_PREFIX
                       <span class="hljs-built_in"> Prefix </span><span class="hljs-keyword">for</span> the dataset you would like <span class="hljs-keyword">to</span> test against
                        your reference model. Remember, the model will <span class="hljs-keyword">not</span>
                        function well <span class="hljs-keyword">if</span> it does <span class="hljs-keyword">not</span> include the same
                        features, <span class="hljs-keyword">and</span> these features should be on the same
                        numeric scale, you can leave off the <span class="hljs-string">'.dataForML.h5'</span>
                        suffix.
  --ref_model_prefix REF_MODEL_PREFIX
                       <span class="hljs-built_in"> Prefix </span>of your reference model file, you can leave off
                        the <span class="hljs-string">'.joblib'</span> suffix.
  -v, --verbose         Verbose output.
</code></pre>
<h2><a class="anchor" aria-hidden="true" id="genoml-continuous-supervised-test"></a><a href="#genoml-continuous-supervised-test" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>genoml continuous supervised test</h2>
<pre><code class="hljs">usage: genoml continuous supervised test [-h] [--prefix PREFIX]
                                         [--test_prefix TEST_PREFIX]
                                         [--ref_model_prefix REF_MODEL_PREFIX]
                                         [-v]

optional arguments:
  -h, --help            show this help message <span class="hljs-keyword">and</span> exit
  --prefix<span class="hljs-built_in"> PREFIX </span>     <span class="hljs-built_in"> Prefix </span><span class="hljs-keyword">for</span> your output build.
  --test_prefix TEST_PREFIX
                       <span class="hljs-built_in"> Prefix </span><span class="hljs-keyword">for</span> the dataset you would like <span class="hljs-keyword">to</span> test against
                        your reference model. Remember, the model will <span class="hljs-keyword">not</span>
                        function well <span class="hljs-keyword">if</span> it does <span class="hljs-keyword">not</span> include the same
                        features, <span class="hljs-keyword">and</span> these features should be on the same
                        numeric scale, you can leave off the <span class="hljs-string">'.dataForML.h5'</span>
                        suffix.
  --ref_model_prefix REF_MODEL_PREFIX
                       <span class="hljs-built_in"> Prefix </span>of your reference model file, you can leave off
                        the <span class="hljs-string">'.joblib'</span> suffix.
  -v, --verbose         Verbose output.
</code></pre>
</span></div></article></div><div class="docs-prevnext"><a class="docs-prev button" href="/docs/workflow"><span class="arrow-prev">← </span><span>Previous</span></a></div></div></div><nav class="onPageNav"><ul class="toc-headings"><li><a href="#at-a-glance">At a Glance</a></li><li><a href="#genoml-the-root-of-all-commands">genoml - the root of all commands</a></li><li><a href="#genoml-harmonize">genoml harmonize</a></li><li><a href="#genoml-discrete-supervised-munge">genoml discrete supervised munge</a></li><li><a href="#genoml-continuous-supervised-munge">genoml continuous supervised munge</a></li><li><a href="#genoml-discrete-supervised-train">genoml discrete supervised train</a></li><li><a href="#genoml-continuous-supervised-train">genoml continuous supervised train</a></li><li><a href="#genoml-discrete-supervised-tune">genoml discrete supervised tune</a></li><li><a href="#genoml-continuous-supervised-tune">genoml continuous supervised tune</a></li><li><a href="#genoml-discrete-supervised-test">genoml discrete supervised test</a></li><li><a href="#genoml-continuous-supervised-test">genoml continuous supervised test</a></li></ul></nav></div><footer class="nav-footer" id="footer"><section class="sitemap"><a href="/" class="nav-home"><img src="/img/dna.png" alt="GenoML" width="66" height="58"/></a><div><h5>Docs</h5><a href="/docs/quick-start">Getting Started</a><a href="/docs/contribute">Contributing</a></div><div><h5>Community</h5><a href="/en/users.html">Users</a></div><div><h5>More</h5><a href="https://github.com/GenoML/">GitHub</a><a class="github-button" href="https://github.com/GenoML/genoml" data-icon="octicon-star" data-count-href="/GenoML/genoml/stargazers" data-show-count="true" data-count-aria-label="# stargazers on GitHub" aria-label="Star this project on GitHub">Star</a></div></section><section class="copyright">Copyright © 2020 GenoML</section></footer></div></body></html>