<!DOCTYPE html><html lang="en"><head><meta charSet="utf-8"/><meta http-equiv="X-UA-Compatible" content="IE=edge"/><title>The Command Line Interface (CLI) · GenoML</title><meta name="viewport" content="width=device-width"/><meta name="generator" content="Docusaurus"/><meta name="description" content="&lt;p&gt;Detailed command line options for &lt;code&gt;genoml-train&lt;/code&gt;, &lt;code&gt;genoml-inference&lt;/code&gt;, and &lt;code&gt;genoml-cli&lt;/code&gt;. Please note &lt;code&gt;genoml-cli&lt;/code&gt; is only recommended for debugging and advanced users:&lt;/p&gt;
"/><meta name="docsearch:language" content="en"/><meta property="og:title" content="The Command Line Interface (CLI) · GenoML"/><meta property="og:type" content="website"/><meta property="og:url" content="https://genoml.github.io/index.html"/><meta property="og:description" content="&lt;p&gt;Detailed command line options for &lt;code&gt;genoml-train&lt;/code&gt;, &lt;code&gt;genoml-inference&lt;/code&gt;, and &lt;code&gt;genoml-cli&lt;/code&gt;. Please note &lt;code&gt;genoml-cli&lt;/code&gt; is only recommended for debugging and advanced users:&lt;/p&gt;
"/><meta property="og:image" content="https://genoml.github.io/img/docusaurus.png"/><meta name="twitter:card" content="summary"/><meta name="twitter:image" content="https://genoml.github.io/img/docusaurus.png"/><link rel="shortcut icon" href="/img/favicon.png"/><link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css"/><script>
              (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
              (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
              m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
              })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

              ga('create', 'UA-134657372-1', 'auto');
              ga('send', 'pageview');
            </script><script type="text/javascript" src="https://buttons.github.io/buttons.js"></script><link rel="stylesheet" href="/css/main.css"/></head><body class="sideNavVisible separateOnPageNav"><div class="fixedHeaderContainer"><div class="headerWrapper wrapper"><header><a href="/"><img class="logo" src="/img/dna.png" alt="GenoML"/><h2 class="headerTitleWithLogo">GenoML</h2></a><div class="navigationWrapper navigationSlider"><nav class="slidingNav"><ul class="nav-site nav-site-internal"><li class="siteNavGroupActive"><a href="/docs/about" target="_self">About GenoML</a></li><li class="siteNavGroupActive"><a href="/docs/quick-start" target="_self">Getting Started</a></li><li class="siteNavGroupActive"><a href="/docs/help" target="_self">Help</a></li><li class=""><a href="https://github.com/GenoML" target="_self">GitHub</a></li></ul></nav></div></header></div></div><div class="navPusher"><div class="docMainWrapper wrapper"><div class="container docsNavContainer" id="docsNav"><nav class="toc"><div class="toggleNav"><section class="navWrapper wrapper"><div class="navBreadcrumb wrapper"><div class="navToggle" id="navToggler"><i></i></div><h2><i>›</i><span>API Reference</span></h2><div class="tocToggler" id="tocToggler"><i class="icon-toc"></i></div></div><div class="navGroups"><div class="navGroup"><h3 class="navGroupCategoryTitle">Intro</h3><ul><li class="navListItem"><a class="navItem" href="/docs/about">About GenoML</a></li><li class="navListItem"><a class="navItem" href="/docs/help">Help</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">Getting Started</h3><ul><li class="navListItem"><a class="navItem" href="/docs/quick-start">Quick start</a></li><li class="navListItem"><a class="navItem" href="/docs/usage">Usage</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">API Reference</h3><ul><li class="navListItem navListItemActive"><a class="navItem" href="/docs/cli">The Command Line Interface (CLI)</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">Contributing </h3><ul><li class="navListItem"><a class="navItem" href="/docs/contribute">Developing GenoML</a></li></ul></div></div></section></div><script>
            document.addEventListener('DOMContentLoaded', function() {
              createToggler('#navToggler', '#docsNav', 'docsSliderActive');
              createToggler('#tocToggler', 'body', 'tocActive');

              const headings = document.querySelector('.toc-headings');
              headings && headings.addEventListener('click', function(event) {
                if (event.target.tagName === 'A') {
                  document.body.classList.remove('tocActive');
                }
              }, false);

              function createToggler(togglerSelector, targetSelector, className) {
                var toggler = document.querySelector(togglerSelector);
                var target = document.querySelector(targetSelector);

                if (!toggler) {
                  return;
                }

                toggler.onclick = function(event) {
                  event.preventDefault();

                  target.classList.toggle(className);
                };
              }
            });
        </script></nav></div><div class="container mainContainer"><div class="wrapper"><div class="post"><header class="postHeader"><h1 class="postHeaderTitle">The Command Line Interface (CLI)</h1></header><article><div><span><p>Detailed command line options for <code>genoml-train</code>, <code>genoml-inference</code>, and <code>genoml-cli</code>. Please note <code>genoml-cli</code> is only recommended for debugging and advanced users:</p>
<h2><a class="anchor" aria-hidden="true" id="genoml-train"></a><a href="#genoml-train" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>genoml-train</h2>
<pre><code class="hljs">genoml-train

Usage:
  genoml-train (<span class="hljs-attribute">--geno-prefix</span>=geno_prefix) (<span class="hljs-attribute">--pheno-file</span>=&lt;pheno_file&gt;) (<span class="hljs-attribute">--model-file</span>=model_file) [<span class="hljs-attribute">--addit-file</span>=&lt;addit_file&gt;] [<span class="hljs-attribute">--cov-file</span>=&lt;cov_file&gt;] [<span class="hljs-attribute">--cv-reps</span>=&lt;cv_reps&gt;] [<span class="hljs-attribute">--grid-search</span>=&lt;grid_search&gt;] [<span class="hljs-attribute">--gwas-file</span>=&lt;gwas_file&gt;] [<span class="hljs-attribute">--herit</span>=&lt;herit&gt;] [<span class="hljs-attribute">--impute-data</span>=&lt;impute_data&gt;] [<span class="hljs-attribute">--n-cores</span>=&lt;n_cores&gt;] [<span class="hljs-attribute">--train-speed</span>=&lt;train_speed&gt;] [-v | -vv | -vvv]
  genoml-train -h | --help
  genoml-train --version

Options:
  <span class="hljs-attribute">--geno-prefix</span>=geno_prefix              <span class="hljs-built_in"> Prefix </span>with path <span class="hljs-keyword">to</span> genotype files <span class="hljs-keyword">in</span> PLINK format, *.bed, *.bim <span class="hljs-keyword">and</span> *.fam.
  <span class="hljs-attribute">--pheno-file</span>=&lt;pheno_file&gt;               Path <span class="hljs-keyword">to</span> the phenotype file <span class="hljs-keyword">in</span> PLINK format, *.pheno.
  <span class="hljs-attribute">--model-file</span>=model_file                 Path <span class="hljs-keyword">to</span> the model file <span class="hljs-keyword">to</span> save.
  <span class="hljs-attribute">--gwas-file</span>=&lt;gwas_file&gt;                 Path <span class="hljs-keyword">to</span> the GWAS file, <span class="hljs-keyword">if</span> available.
  <span class="hljs-attribute">--cov-file</span>=&lt;cov_file&gt;                   Path <span class="hljs-keyword">to</span> the covariance file, <span class="hljs-keyword">if</span> available.
  <span class="hljs-attribute">--herit</span>=&lt;herit&gt;                         Heritability estimate of phenotype between 0 <span class="hljs-keyword">and</span> 1, <span class="hljs-keyword">if</span> available.
  <span class="hljs-attribute">--addit-file</span>=&lt;addit_file&gt;               Path <span class="hljs-keyword">to</span> the additional file, <span class="hljs-keyword">if</span> avialable.
  <span class="hljs-attribute">--n-cores</span>=&lt;n_cores&gt;                     Number of cores <span class="hljs-keyword">to</span> be allocated <span class="hljs-keyword">for</span> computation [default: 1].
  <span class="hljs-attribute">--train-speed</span>=&lt;train_speed&gt;             Training speed: (ALL, FAST, FURIOUS, BOOSTED). <span class="hljs-builtin-name">Run</span> all models, only  the fastest models, <span class="hljs-builtin-name">run</span> slightly slower models, <span class="hljs-keyword">or</span> just <span class="hljs-builtin-name">run</span> boosted models which usually perform best when using genotype data [default: BOOSTED].
  <span class="hljs-attribute">--cv-reps</span>=&lt;cv_reps&gt;                     Number of cross-validation. An integer greater than 5. Effects the speed [default: 5].
  <span class="hljs-attribute">--impute-data</span>=&lt;impute_data&gt;             Imputation: (knn, median). Governs secondary imputation <span class="hljs-keyword">and</span> data transformation [default: median].
  <span class="hljs-attribute">--grid-search</span>=&lt;grid_search&gt;             Grid search length <span class="hljs-keyword">for</span> parameters, integer greater than 10, 30 <span class="hljs-keyword">or</span> greater recommended, effects speed of initial tune [default: 10].
  -v -vv -vvv                             Verbose output.
  -h --help                               Show this screen.
  --version                               Show version.

Examples:
  genoml-train <span class="hljs-attribute">--model-file</span>=model.genoml_model <span class="hljs-attribute">--geno-prefix</span>=./exampleData/example <span class="hljs-attribute">--pheno-file</span>=./exampleData/training.pheno <span class="hljs-attribute">--gwas-file</span>=./exampleData/example_GWAS.txt
</code></pre>
<h2><a class="anchor" aria-hidden="true" id="genoml-inference"></a><a href="#genoml-inference" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>genoml-inference</h2>
<pre><code class="hljs">genoml-inference

Usage:
  genoml-inference (<span class="hljs-attribute">--model-file</span>=model_file) (<span class="hljs-attribute">--valid-dir</span>=valid_dir) (<span class="hljs-attribute">--valid-geno-prefix</span>=valid_geno_prefix) (<span class="hljs-attribute">--valid-pheno-file</span>=&lt;valid_pheno_file&gt;) [<span class="hljs-attribute">--valid-cov-file</span>=&lt;valid_cov_file&gt;] [<span class="hljs-attribute">--valid-addit-file</span>=&lt;valid_addit_file&gt;] [<span class="hljs-attribute">--n-cores</span>=&lt;n_cores&gt;] [-v | -vv | -vvv]
  genoml-inference -h | --help
  genoml-inference --version

Options:
  <span class="hljs-attribute">--model-file</span>=model_file                 Path <span class="hljs-keyword">to</span> the model file.
  <span class="hljs-attribute">--valid-dir</span>=valid_dir                   Folder <span class="hljs-keyword">to</span> save output.
  <span class="hljs-attribute">--n-cores</span>=&lt;n_cores&gt;                     Number of cores <span class="hljs-keyword">to</span> be allocated <span class="hljs-keyword">for</span> computation [default: 1].
  <span class="hljs-attribute">--valid-geno-prefix</span>=valid_geno_prefix  <span class="hljs-built_in"> Prefix </span>with path <span class="hljs-keyword">to</span> the validation genotype files <span class="hljs-keyword">in</span> PLINK format, *.bed, *.bim <span class="hljs-keyword">and</span> *.fam.
  <span class="hljs-attribute">--valid-pheno-file</span>=&lt;valid_pheno_file&gt;   Path <span class="hljs-keyword">to</span> the validation phenotype file <span class="hljs-keyword">in</span> PLINK format, *.pheno.
  <span class="hljs-attribute">--valid-cov-file</span>=&lt;valid_cov_file&gt;       Path <span class="hljs-keyword">to</span> the validation covariance file, <span class="hljs-keyword">if</span> available.
  <span class="hljs-attribute">--valid-addit-file</span>=&lt;valid_addit_file&gt;   Path <span class="hljs-keyword">to</span> the the validation additional file, <span class="hljs-keyword">if</span> avialable.
  -h --help                               Show this screen.
  -v -vv -vvv                             Verbose output.
  --version                               Show version.

Examples:
  genoml-inference <span class="hljs-attribute">--model-file</span>=./model.genoml_model <span class="hljs-attribute">--valid-geno-prefix</span>=./exampleData/validation <span class="hljs-attribute">--valid-pheno-file</span>=./exampleData/validation.pheno
</code></pre>
<h2><a class="anchor" aria-hidden="true" id="genoml-cli"></a><a href="#genoml-cli" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>genoml-cli</h2>
<pre><code class="hljs">genoml-cli

Usage:
  genoml-cli data-prune  (<span class="hljs-attribute">--geno-prefix</span>=geno_prefix) (<span class="hljs-attribute">--pheno-file</span>=&lt;pheno_file&gt;) [<span class="hljs-attribute">--gwas-file</span>=&lt;gwas_file&gt;] [<span class="hljs-attribute">--cov-file</span>=&lt;cov_file&gt;] [<span class="hljs-attribute">--herit</span>=&lt;herit&gt;] [<span class="hljs-attribute">--addit-file</span>=&lt;addit_file&gt;] [<span class="hljs-attribute">--temp-dir</span>=&lt;directory&gt;] [<span class="hljs-attribute">--prune-prefix</span>=&lt;prune_prefix&gt;]
  genoml-cli model-train (<span class="hljs-attribute">--prune-prefix</span>=prune_prefix) (<span class="hljs-attribute">--pheno-file</span>=&lt;pheno_file&gt;) [<span class="hljs-attribute">--n-cores</span>=&lt;n_cores&gt;] [<span class="hljs-attribute">--train-speed</span>=&lt;train_speed&gt;] [<span class="hljs-attribute">--cv-reps</span>=&lt;cv_reps&gt;] [<span class="hljs-attribute">--grid-search</span>=&lt;grid_search&gt;] [<span class="hljs-attribute">--impute-data</span>=&lt;impute_data&gt;]
  genoml-cli model-tune (<span class="hljs-attribute">--prune-prefix</span>=prune_prefix) (<span class="hljs-attribute">--pheno-file</span>=&lt;pheno_file&gt;) [<span class="hljs-attribute">--cv-reps</span>=&lt;cv_reps&gt;] [<span class="hljs-attribute">--grid-search</span>=&lt;grid_search&gt;] [<span class="hljs-attribute">--impute-data</span>=&lt;impute_data&gt;] [<span class="hljs-attribute">--best-model-name</span>=&lt;best_model_name&gt;]
  genoml-cli model-validate (<span class="hljs-attribute">--prune-prefix</span>=prune_prefix) (<span class="hljs-attribute">--pheno-file</span>=&lt;pheno_file&gt;) (<span class="hljs-attribute">--geno-prefix</span>=geno_prefix) (<span class="hljs-attribute">--valid-geno-prefix</span>=valid_geno_prefix) (<span class="hljs-attribute">--valid-pheno-file</span>=&lt;valid_pheno_file&gt;) [<span class="hljs-attribute">--valid-cov-file</span>=&lt;valid_cov_file&gt;] [<span class="hljs-attribute">--gwas-file</span>=&lt;gwas_file&gt;] [<span class="hljs-attribute">--valid-addit-file</span>=&lt;valid_addit_file&gt;] [<span class="hljs-attribute">--n-cores</span>=&lt;n_cores&gt;] [<span class="hljs-attribute">--impute-data</span>=&lt;impute_data&gt;]  [<span class="hljs-attribute">--best-model-name</span>=&lt;best_model_name&gt;]
  genoml-cli -h | --help
  genoml-cli --version

Options:
  <span class="hljs-attribute">--geno-prefix</span>=geno_prefix              <span class="hljs-built_in"> Prefix </span>with path <span class="hljs-keyword">to</span> genotype files <span class="hljs-keyword">in</span> PLINK format, *.bed, *.bim <span class="hljs-keyword">and</span> *.fam.
  <span class="hljs-attribute">--pheno-file</span>=&lt;pheno_file&gt;               Path <span class="hljs-keyword">to</span> the phenotype file <span class="hljs-keyword">in</span> PLINK format, *.pheno.
  <span class="hljs-attribute">--gwas-file</span>=&lt;gwas_file&gt;                 Path <span class="hljs-keyword">to</span> the GWAS file, <span class="hljs-keyword">if</span> available.
  <span class="hljs-attribute">--cov-file</span>=&lt;cov_file&gt;                   Path <span class="hljs-keyword">to</span> the covariance file, <span class="hljs-keyword">if</span> available.
  <span class="hljs-attribute">--herit</span>=&lt;herit&gt;                         Heritability estimate of phenotype between 0 <span class="hljs-keyword">and</span> 1, <span class="hljs-keyword">if</span> available.
  <span class="hljs-attribute">--addit-file</span>=&lt;addit_file&gt;               Path <span class="hljs-keyword">to</span> the additional file, <span class="hljs-keyword">if</span> avialable.
  <span class="hljs-attribute">--temp-dir</span>=&lt;directory&gt;                  Directory <span class="hljs-keyword">for</span> temporary files [default: ./tmp/].
  <span class="hljs-attribute">--n-cores</span>=&lt;n_cores&gt;                     Number of cores <span class="hljs-keyword">to</span> be allocated <span class="hljs-keyword">for</span> computation [default: 1].
  <span class="hljs-attribute">--prune-prefix</span>=prune_prefix            <span class="hljs-built_in"> Prefix </span>given <span class="hljs-keyword">to</span> you at the end of pruning stage.
  <span class="hljs-attribute">--train-speed</span>=&lt;train_speed&gt;             Training speed: (ALL, FAST, FURIOUS, BOOSTED). <span class="hljs-builtin-name">Run</span> all models, only  the fastest models, <span class="hljs-builtin-name">run</span> slightly slower models, <span class="hljs-keyword">or</span> just <span class="hljs-builtin-name">run</span> boosted models which usually perform best when using genotype data [default: BOOSTED].
  <span class="hljs-attribute">--cv-reps</span>=&lt;cv_reps&gt;                     Number of cross-validation. An integer greater than 5. Effects the speed [default: 5].
  <span class="hljs-attribute">--impute-data</span>=&lt;impute_data&gt;             Imputation: (knn, median). Governs secondary imputation <span class="hljs-keyword">and</span> data transformation [default: median].
  <span class="hljs-attribute">--grid-search</span>=&lt;grid_search&gt;             Grid search length <span class="hljs-keyword">for</span> parameters, integer greater than 10, 30 <span class="hljs-keyword">or</span> greater recommended, effects speed of initial tune [default: 10].
  <span class="hljs-attribute">--best-model-name</span>=&lt;best_model_name&gt;     Name <span class="hljs-keyword">for</span> the best model [default: best_model].
  <span class="hljs-attribute">--valid-geno-prefix</span>=valid_geno_prefix  <span class="hljs-built_in"> Prefix </span>with path <span class="hljs-keyword">to</span> the validation genotype files <span class="hljs-keyword">in</span> PLINK format, *.bed, *.bim <span class="hljs-keyword">and</span> *.fam.
  <span class="hljs-attribute">--valid-pheno-file</span>=&lt;valid_pheno_file&gt;   Path <span class="hljs-keyword">to</span> the validation phenotype file <span class="hljs-keyword">in</span> PLINK format, *.pheno.
  <span class="hljs-attribute">--valid-cov-file</span>=&lt;valid_cov_file&gt;       Path <span class="hljs-keyword">to</span> the validation covariance file, <span class="hljs-keyword">if</span> available.
  <span class="hljs-attribute">--valid-addit-file</span>=&lt;valid_addit_file&gt;   Path <span class="hljs-keyword">to</span> the the validation additional file, <span class="hljs-keyword">if</span> avialable.
  -h --help                               Show this screen.
  --version                               Show version.

Examples:
  genoml-cli data-prune <span class="hljs-attribute">--geno-prefix</span>=./exampleData/example <span class="hljs-attribute">--pheno-file</span>=./exampleData/training.pheno
  genoml-cli data-prune <span class="hljs-attribute">--geno-prefix</span>=./exampleData/example <span class="hljs-attribute">--pheno-file</span>=./exampleData/training.pheno  <span class="hljs-attribute">--gwas-file</span>=./exampleData/example_GWAS.txt
  genoml-cli data-prune <span class="hljs-attribute">--geno-prefix</span>=./exampleData/example <span class="hljs-attribute">--pheno-file</span>=./exampleData/training.pheno <span class="hljs-attribute">--cov-file</span>=./exampleData/training.cov <span class="hljs-attribute">--gwas-file</span>=./exampleData/example_GWAS.txt <span class="hljs-attribute">--addit-file</span>=./exampleData/training.addit
  genoml-cli data-prune <span class="hljs-attribute">--geno-prefix</span>=./exampleData/example <span class="hljs-attribute">--pheno-file</span>=./exampleData/training.pheno  <span class="hljs-attribute">--gwas-file</span>=./exampleData/example_GWAS.txt <span class="hljs-attribute">--addit-file</span>=./exampleData/training.addit <span class="hljs-attribute">--herit</span>=0.2
  genoml-cli data-prune <span class="hljs-attribute">--geno-prefix</span>=./exampleData/example <span class="hljs-attribute">--pheno-file</span>=./exampleData/training.pheno <span class="hljs-attribute">--cov-file</span>=./exampleData/training.cov <span class="hljs-attribute">--gwas-file</span>=./exampleData/example_GWAS.txt <span class="hljs-attribute">--addit-file</span>=./exampleData/training.addit <span class="hljs-attribute">--herit</span>=0.5
  genoml-cli model-train <span class="hljs-attribute">--prune-prefix</span>=./tmp/20181225-230052 <span class="hljs-attribute">--pheno-file</span>=./exampleData/training.pheno
  genoml-cli model-tune <span class="hljs-attribute">--prune-prefix</span>=./tmp/20181225-230052 <span class="hljs-attribute">--pheno-file</span>=./exampleData/training.pheno
  genoml-cli model-validate <span class="hljs-attribute">--prune-prefix</span>=./tmp/20181225-230052 <span class="hljs-attribute">--pheno-file</span>=./exampleData/training.pheno <span class="hljs-attribute">--valid-geno-prefix</span>=./exampleData/validation <span class="hljs-attribute">--valid-pheno-file</span>=./exampleData/validation.pheno
</code></pre>
</span></div></article></div><div class="docs-prevnext"><a class="docs-prev button" href="/docs/Workflow"><span class="arrow-prev">← </span><span>Previous</span></a><a class="docs-next button" href="/docs/contribute"><span class="function-name-prevnext">Developing GenoML</span><span class="arrow-next"> →</span></a></div></div></div><nav class="onPageNav"><ul class="toc-headings"><li><a href="#genoml-train">genoml-train</a></li><li><a href="#genoml-inference">genoml-inference</a></li><li><a href="#genoml-cli">genoml-cli</a></li></ul></nav></div><footer class="nav-footer" id="footer"><section class="sitemap"><a href="/" class="nav-home"><img src="/img/dna.png" alt="GenoML" width="66" height="58"/></a><div><h5>Docs</h5><a href="/docs/en/overview.html">Getting Started</a><a href="/docs/en/contribute.html">Contributing</a></div><div><h5>Community</h5><a href="/en/users.html">Users</a></div><div><h5>More</h5><a href="https://github.com/">GitHub</a><a class="github-button" data-icon="octicon-star" data-count-href="/genoml/genoml-cli/stargazers" data-show-count="true" data-count-aria-label="# stargazers on GitHub" aria-label="Star this project on GitHub">Star</a></div></section><section class="copyright">Copyright © 2019 GenoML</section></footer></div></body></html>