<!DOCTYPE html><html lang="en"><head><meta charSet="utf-8"/><meta http-equiv="X-UA-Compatible" content="IE=edge"/><title>Quick start · GenoML</title><meta name="viewport" content="width=device-width"/><meta name="generator" content="Docusaurus"/><meta name="description" content="&lt;h1&gt;&lt;a class=&quot;anchor&quot; aria-hidden=&quot;true&quot; id=&quot;genoml&quot;&gt;&lt;/a&gt;&lt;a href=&quot;#genoml&quot; aria-hidden=&quot;true&quot; class=&quot;hash-link&quot;&gt;&lt;svg class=&quot;hash-link-icon&quot; aria-hidden=&quot;true&quot; height=&quot;16&quot; version=&quot;1.1&quot; viewBox=&quot;0 0 16 16&quot; width=&quot;16&quot;&gt;&lt;path fill-rule=&quot;evenodd&quot; d=&quot;M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z&quot;&gt;&lt;/path&gt;&lt;/svg&gt;&lt;/a&gt;GenoML&lt;/h1&gt;
"/><meta name="docsearch:language" content="en"/><meta property="og:title" content="Quick start · GenoML"/><meta property="og:type" content="website"/><meta property="og:url" content="https://genoml.github.io/index.html"/><meta property="og:description" content="&lt;h1&gt;&lt;a class=&quot;anchor&quot; aria-hidden=&quot;true&quot; id=&quot;genoml&quot;&gt;&lt;/a&gt;&lt;a href=&quot;#genoml&quot; aria-hidden=&quot;true&quot; class=&quot;hash-link&quot;&gt;&lt;svg class=&quot;hash-link-icon&quot; aria-hidden=&quot;true&quot; height=&quot;16&quot; version=&quot;1.1&quot; viewBox=&quot;0 0 16 16&quot; width=&quot;16&quot;&gt;&lt;path fill-rule=&quot;evenodd&quot; d=&quot;M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z&quot;&gt;&lt;/path&gt;&lt;/svg&gt;&lt;/a&gt;GenoML&lt;/h1&gt;
"/><meta property="og:image" content="https://genoml.github.io/img/docusaurus.png"/><meta name="twitter:card" content="summary"/><meta name="twitter:image" content="https://genoml.github.io/img/docusaurus.png"/><link rel="shortcut icon" href="/img/favicon.png"/><link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css"/><script>
              (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
              (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
              m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
              })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

              ga('create', 'UA-134657372-1', 'auto');
              ga('send', 'pageview');
            </script><script type="text/javascript" src="https://buttons.github.io/buttons.js"></script><link rel="stylesheet" href="/css/main.css"/><script src="/js/codetabs.js"></script></head><body class="sideNavVisible separateOnPageNav"><div class="fixedHeaderContainer"><div class="headerWrapper wrapper"><header><a href="/"><img class="logo" src="/img/dna.png" alt="GenoML"/><h2 class="headerTitleWithLogo">GenoML</h2></a><div class="navigationWrapper navigationSlider"><nav class="slidingNav"><ul class="nav-site nav-site-internal"><li class="siteNavGroupActive"><a href="/docs/introduction" target="_self">About</a></li><li class="siteNavGroupActive siteNavItemActive"><a href="/docs/quick-start" target="_self">Getting Started</a></li><li class="siteNavGroupActive"><a href="/docs/contribute" target="_self">Contribute</a></li><li class=""><a href="https://github.com/GenoML" target="_self">GitHub</a></li></ul></nav></div></header></div></div><div class="navPusher"><div class="docMainWrapper wrapper"><div class="container docsNavContainer" id="docsNav"><nav class="toc"><div class="toggleNav"><section class="navWrapper wrapper"><div class="navBreadcrumb wrapper"><div class="navToggle" id="navToggler"><i></i></div><h2><i>›</i><span>Getting Started</span></h2><div class="tocToggler" id="tocToggler"><i class="icon-toc"></i></div></div><div class="navGroups"><div class="navGroup"><h3 class="navGroupCategoryTitle">GenoML</h3><ul class=""><li class="navListItem"><a class="navItem" href="/docs/introduction">Introduction</a></li><li class="navListItem"><a class="navItem" href="/docs/principles">Principles and Philosophy</a></li><li class="navListItem"><a class="navItem" href="/docs/roadmap">Roadmap</a></li><li class="navListItem"><a class="navItem" href="/docs/contribute">Contributing</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">Getting Started</h3><ul class=""><li class="navListItem navListItemActive"><a class="navItem" href="/docs/quick-start">Quick start</a></li><li class="navListItem"><a class="navItem" href="/docs/usage">Usage</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">How it works</h3><ul class=""><li class="navListItem"><a class="navItem" href="/docs/workflow">Workflow</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">API Reference</h3><ul class=""><li class="navListItem"><a class="navItem" href="/docs/cli">The Command Line Interface (CLI)</a></li></ul></div></div></section></div><script>
            var coll = document.getElementsByClassName('collapsible');
            var checkActiveCategory = true;
            for (var i = 0; i < coll.length; i++) {
              var links = coll[i].nextElementSibling.getElementsByTagName('*');
              if (checkActiveCategory){
                for (var j = 0; j < links.length; j++) {
                  if (links[j].classList.contains('navListItemActive')){
                    coll[i].nextElementSibling.classList.toggle('hide');
                    coll[i].childNodes[1].classList.toggle('rotate');
                    checkActiveCategory = false;
                    break;
                  }
                }
              }

              coll[i].addEventListener('click', function() {
                var arrow = this.childNodes[1];
                arrow.classList.toggle('rotate');
                var content = this.nextElementSibling;
                content.classList.toggle('hide');
              });
            }

            document.addEventListener('DOMContentLoaded', function() {
              createToggler('#navToggler', '#docsNav', 'docsSliderActive');
              createToggler('#tocToggler', 'body', 'tocActive');

              const headings = document.querySelector('.toc-headings');
              headings && headings.addEventListener('click', function(event) {
                if (event.target.tagName === 'A') {
                  document.body.classList.remove('tocActive');
                }
              }, false);

              function createToggler(togglerSelector, targetSelector, className) {
                var toggler = document.querySelector(togglerSelector);
                var target = document.querySelector(targetSelector);

                if (!toggler) {
                  return;
                }

                toggler.onclick = function(event) {
                  event.preventDefault();

                  target.classList.toggle(className);
                };
              }
            });
        </script></nav></div><div class="container mainContainer"><div class="wrapper"><div class="post"><header class="postHeader"><h1 class="postHeaderTitle">Quick start</h1></header><article><div><span><h1><a class="anchor" aria-hidden="true" id="genoml"></a><a href="#genoml" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>GenoML</h1>
<p align="center">
  <img width="300" height="300" src="GenoML2_white-04.png">
</p>
<h1><a class="anchor" aria-hidden="true" id="how-to-get-started-with-genoml"></a><a href="#how-to-get-started-with-genoml" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>How to Get Started with GenoML</h1>
<h3><a class="anchor" aria-hidden="true" id="introduction"></a><a href="#introduction" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Introduction</h3>
<p>GenoML is an Automated Machine Learning (AutoML) for genomics data. In general, use a Linux or Mac with Python &gt;3.5 for best results. <strong>This repository is under active development!</strong></p>
<p>This README is a brief look into how to structure arguments and what arguments are available at each phase for the GenoML CLI.</p>
<h3><a class="anchor" aria-hidden="true" id="installing-downloading-example-data"></a><a href="#installing-downloading-example-data" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Installing + Downloading Example Data</h3>
<ul>
<li>Install using pip</li>
</ul>
<p><code>Coming Soon!</code></p>
<ul>
<li>To install the <code>examples/</code> directory (~315 KB), you can use SVN (pre-installed on Macs)</li>
</ul>
<p><code>svn checkout https://github.com/GenoML/genoml.git/branches/python_v1.5/examples</code></p>
<p>... or download the data from <a href="https://github.com/GenoML/genoml/tree/python_v1.5/examples">here</a>.</p>
<h3><a class="anchor" aria-hidden="true" id="table-of-contents"></a><a href="#table-of-contents" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Table of Contents</h3>
<h4><a class="anchor" aria-hidden="true" id="0-optional-how-to-set-up-a-virtual-environment-via-conda-0"></a><a href="#0-optional-how-to-set-up-a-virtual-environment-via-conda-0" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a><a href="#0">0. (OPTIONAL) How to Set Up a Virtual Environment via Conda</a></h4>
<h4><a class="anchor" aria-hidden="true" id="1-munging-with-genoml-1"></a><a href="#1-munging-with-genoml-1" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a><a href="#1">1. Munging with GenoML</a></h4>
<h4><a class="anchor" aria-hidden="true" id="2-training-with-genoml-2"></a><a href="#2-training-with-genoml-2" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a><a href="#2">2. Training with GenoML</a></h4>
<h4><a class="anchor" aria-hidden="true" id="3-tuning-with-genoml-3"></a><a href="#3-tuning-with-genoml-3" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a><a href="#3">3. Tuning with GenoML</a></h4>
<h4><a class="anchor" aria-hidden="true" id="4-testing-validating-with-genoml-4"></a><a href="#4-testing-validating-with-genoml-4" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a><a href="#4">4. Testing/Validating with GenoML</a></h4>
<h4><a class="anchor" aria-hidden="true" id="5-experimental-features-5"></a><a href="#5-experimental-features-5" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a><a href="#5">5. Experimental Features</a></h4>
<p><a id="0"></a></p>
<h2><a class="anchor" aria-hidden="true" id="0-optional-how-to-set-up-a-virtual-environment-via-conda"></a><a href="#0-optional-how-to-set-up-a-virtual-environment-via-conda" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>0. [OPTIONAL] How to Set Up a Virtual Environment via Conda</h2>
<p>You can create a virtual environment to run GenoML, if you prefer.
If you already have the Anaconda Distribution, this is fairly simple.</p>
<p>To create and activate a virtual environment:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> To create a virtual environment</span>
conda create -n GenoML python=3.7
<span class="hljs-meta">
#</span><span class="bash"> To activate a virtual environment</span>
conda activate GenoML
<span class="hljs-meta">
#</span><span class="bash"><span class="hljs-comment"># MISC</span></span>
<span class="hljs-meta">#</span><span class="bash"> To deactivate the virtual environment</span>
<span class="hljs-meta">#</span><span class="bash"> conda deactivate GenoML</span>
<span class="hljs-meta">
#</span><span class="bash"> To delete your virtual environment</span>
<span class="hljs-meta">#</span><span class="bash"> conda env remove -n GenoML</span>
</code></pre>
<p>To install the package at this path:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Install the package at this path</span>
pip install .
<span class="hljs-meta">
#</span><span class="bash"> MISC</span>
    # To save out the environment requirements to a .txt file
<span class="hljs-meta">#</span><span class="bash"> pip freeze &gt; requirements.txt</span>

    # Removing a conda virtualenv
<span class="hljs-meta">#</span><span class="bash"> conda remove --name GenoML --all </span>
</code></pre>
<p><a id="1"></a></p>
<h2><a class="anchor" aria-hidden="true" id="1-munging-with-genoml"></a><a href="#1-munging-with-genoml" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>1. Munging with GenoML</h2>
<p>Munging with GenoML will, at minimum, do the following:</p>
<ul>
<li>Prune your genotypes using PLINK v1.9 (if <code>--geno</code> flag is used). You don't have to include genotypes, we just generally do. At its core, GenoML is a fully functional auto-ML. After all, GenoML = <strong>Geno</strong>mics + <strong>M</strong>achine <strong>L</strong>earning.</li>
<li>Impute per column using median or mean (can be changed with the <code>--impute</code> flag)</li>
<li>Z-scaling of features and removing columns with a std dev = 0</li>
</ul>
<p><strong>Required</strong> arguments for GenoML munging are <code>--prefix</code> and <code>--pheno</code></p>
<ul>
<li><code>data</code> : Is the data <code>continuous</code> or <code>discrete</code>?</li>
<li><code>method</code>: Do you want to use <code>supervised</code> or <code>unsupervised</code> machine learning? <em>(unsupervised currently under development)</em></li>
<li><code>mode</code>:  would you like to <code>munge</code>, <code>train</code>, <code>tune</code>, or <code>test</code> your model?</li>
<li><code>--prefix</code> : Where would you like your outputs to be saved?</li>
<li><code>--pheno</code> : Where is your phenotype file? This file only has 2 columns, ID in one, and PHENO in the other (0 for controls and 1 for cases if you are using a <strong>discrete</strong> outcome to indicate disease status for example ... if you are just using a <strong>continuous</strong> outcome, just make sure it is numeric although we generally z scale all continuous data to a mean of zero and a standard deviation of one).</li>
</ul>
<p>Be sure to have your files formatted the same as the examples, key points being:</p>
<ul>
<li><strong>0=controls and 1=case</strong> in your phenotype file</li>
<li>Your phenotype file consisting <strong>only</strong> of the &quot;ID&quot; and &quot;PHENO&quot; columns</li>
<li>Your sample IDs matching across all files</li>
<li>Your sample IDs not consisting with only integers (add a prefix or suffix to all sample IDs ensuring they are alphanumeric if this is the case prior to running GenoML)</li>
</ul>
<blockquote>
<p><em>Note:</em> The following examples are for discrete data, but if you substitute following commands with <code>continuous</code> instead of discrete, you can preprocess your continuous data! Also note that GenoML only likes numbers! Outside of the ID columns, GenoML hates strings! If you have a categorical variable you want to include as a feature, please use one-hot encoding to make each category its own discrete feature.</p>
</blockquote>
<p>If you would like to munge just with genotypes (in PLINK binary format), the simplest command is the following:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML munging on discrete data using PLINK genotype binary files and a phenotype file </span>

genoml discrete supervised munge \
--prefix outputs/test_discrete_geno \
--geno examples/discrete/training \
--pheno examples/discrete/training_pheno.csv
</code></pre>
<p>You can choose to impute on <code>mean</code> or <code>median</code> by modifying the <code>--impute</code> flag, like so <em>(default is median)</em>:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML munging on discrete data using PLINK genotype binary files and a phenotype file and specifying impute</span>

genoml discrete supervised munge \
--prefix outputs/test_discrete_geno \
--geno examples/discrete/training \
--pheno examples/discrete/training_pheno.csv \
--impute mean
</code></pre>
<p>If you suspect collinear variables, and think this will be a problem for training the model moving forward, you can use <a href="https://www.investopedia.com/terms/v/variance-inflation-factor.asp">variance inflation factor</a> (VIF) filtering:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML munging on discrete data using PLINK genotype binary files and a phenotype file <span class="hljs-keyword">while</span> using VIF to remove multicollinearity </span>

genoml discrete supervised munge \
--prefix outputs/test_discrete_geno \
--geno examples/discrete/training \
--pheno examples/discrete/training_pheno.csv \
--vif 5 \
--iter 1
</code></pre>
<ul>
<li>The <code>--vif</code> flag specifies the VIF threshold you would like to use (5 is recommended)</li>
<li>The number of iterations you'd like to run can be modified with the <code>--iter</code> flag (if you have or anticipate many collinear variables, it's a good idea to increase the iterations)</li>
<li>The VIF calcs themselves can be resource intensive, just a warning. Feature selection via extraTrees might be a better option and is discussed later on</li>
</ul>
<p>Well, what if you had GWAS summary statistics handy, and would like to just use the same SNPs outlined in that file? You can do so by running the following:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML munging on discrete data using PLINK genotype binary files, a phenotype file, and a GWAS summary statistics file </span>

genoml discrete supervised munge \
--prefix outputs/test_discrete_geno \
--geno examples/discrete/training \
--pheno examples/discrete/training_pheno.csv \
--gwas examples/discrete/example_GWAS.csv
</code></pre>
<blockquote>
<p><em>Note:</em> When using the GWAS flag, the PLINK binaries will be pruned to include matching SNPs to the GWAS file.</p>
</blockquote>
<p>...and if you wanted to add a p-value cut-off...</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML munging on discrete data using PLINK genotype binary files, a phenotype file, and a GWAS summary statistics file with a p-value cut-off </span>

genoml discrete supervised munge \
--prefix outputs/test_discrete_geno \
--geno examples/discrete/training \
--pheno examples/discrete/training_pheno.csv \
--gwas examples/discrete/example_GWAS.csv \
--p 0.01
</code></pre>
<p>Do you have additional data you would like to incorporate? Perhaps clinical, demographic, or transcriptomics data? If coded and all numerical, these can be added as an <code>--addit</code> file by doing the following:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML munging on discrete data using PLINK genotype binary files, a phenotype file, and an addit file</span>

genoml discrete supervised munge \
--prefix outputs/test_discrete_geno \
--geno examples/discrete/training \
--pheno examples/discrete/training_pheno.csv \
--addit examples/discrete/training_addit.csv
</code></pre>
<p>You also have the option of not using PLINK binary files if you would like to just preprocess (and then, later train) on a phenotype and addit file by doing the following:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML munging on discrete data using PLINK genotype binary files, a phenotype file, and an addit file</span>

genoml discrete supervised munge \
--prefix outputs/test_discrete_geno \
--pheno examples/discrete/training_pheno.csv \
--addit examples/discrete/training_addit.csv
</code></pre>
<p>Are you interested in selecting and ranking your features? If so, you can use the <code>--feature_selection</code> flag and specify like so...:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML munging on discrete data using PLINK genotype binary files, a phenotype file, and running feature selection </span>

genoml discrete supervised munge \
--prefix outputs/test_discrete_geno \
--geno examples/discrete/training \
--pheno examples/discrete/training_pheno.csv \
--addit examples/discrete/training_addit.csv \
--feature_selection 50
</code></pre>
<p>The <code>--feature_selection</code> flag uses extraTrees (<a href="https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.ExtraTreesClassifier.html">classifier</a> for discrete data; <a href="https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.ExtraTreesRegressor.html">regressor</a> for continuous data) to output a <code>*_approx_feature_importance.txt</code> file with the features most contributing to your model at the top. <strong>We generally reccommend feature selection!</strong></p>
<p><a id="2"></a></p>
<h2><a class="anchor" aria-hidden="true" id="2-training-with-genoml"></a><a href="#2-training-with-genoml" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>2. Training with GenoML</h2>
<p>Training with GenoML competes a number of different algorithms and outputs the best algorithm based on a specific metric that can be tweaked using the <code>--metric_max</code> flag <em>(default is AUC)</em>. These algorithms are pretty common and were selected based on aggregate <a href="https://www.kaggle.com/">kaggle</a> performance and a survey of ML researchers in the healthcare space.</p>
<p><strong>Required</strong> arguments for GenoML are the following:</p>
<ul>
<li><code>data</code> : Is the data <code>continuous</code> or <code>discrete</code>?</li>
<li><code>method</code>: Do you want to use <code>supervised</code> or <code>unsupervised</code> machine learning? <em>(unsupervised currently under development)</em></li>
<li><code>mode</code>:  would you like to <code>munge</code>, <code>train</code>, <code>tune</code>, or <code>test</code> your model?</li>
<li><code>--prefix</code> : Where would you like your outputs to be saved?</li>
</ul>
<p>The most basic command to train your model looks like the following, it looks for the <code>*.dataForML</code> file that was generated in the munging step:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML supervised training after munging on discrete data</span>

genoml discrete supervised train \
--prefix outputs/test_discrete_geno
</code></pre>
<p>If you would like to determine the best competing algorithm by something other than the AUC, you can do so by changing the <code>--metric_max</code> flag (Options include <code>AUC</code>, <code>Balanced_Accuracy</code>, <code>Sensitivity</code>, and <code>Specificity</code>) :</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML supervised training after munging on discrete data and specifying the metric to maximize by </span>

genoml discrete supervised train \
--prefix outputs/test_discrete_geno \
--metric_max Sensitivity
</code></pre>
<blockquote>
<p><em>Note:</em> The <code>--metric_max</code> flag is only available for discrete datasets.</p>
</blockquote>
<p><a id="3"></a></p>
<h2><a class="anchor" aria-hidden="true" id="3-tuning-with-genoml"></a><a href="#3-tuning-with-genoml" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>3. Tuning with GenoML</h2>
<p>The most basic command to tune your model looks like the following, it looks for the file that was generated in the training step:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML supervised tuning after munging and training on discrete data</span>

genoml discrete supervised tune \
--prefix outputs/test_discrete_geno
</code></pre>
<p>If you are interested in changing the number of iterations the tuning process goes through by modifying <code>--max_tune</code> <em>(default is 50)</em>, or the number of cross-validations by modifying <code>--n_cv</code> <em>(default is 5)</em>, this is what the command would look like:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML supervised tuning after munging and training on discrete data, modifying the number of iterations and cross-validations </span>

genoml discrete supervised tune \
--prefix outputs/test_discrete_geno \
--max_tune 10 --n_cv 3
</code></pre>
<p>If you are interested in tuning on another metric other than AUC <em>(default is AUC)</em>, you can modify <code>--metric_tune</code> (options are <code>AUC</code> or <code>Balanced_Accuracy</code>) by doing the following:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML supervised tuning after munging and training on discrete data, modifying the metric to tune by</span>

genoml discrete supervised tune \
--prefix outputs/test_discrete_geno \
--metric_tune Balanced_Accuracy
</code></pre>
<p><a id="4"></a></p>
<h2><a class="anchor" aria-hidden="true" id="4-testing-validation-with-genoml"></a><a href="#4-testing-validation-with-genoml" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>4. Testing/Validation with GenoML</h2>
<p>In order to properly test how your model performs on a dataset it's never seen before (but you start with different PLINK binaries), we have created the harmonization step that will:</p>
<ol>
<li>Keep only the same SNPs between your reference dataset and the dataset you are using for validation</li>
<li>Force the reference alleles in the validation dataset to match your reference dataset</li>
<li>Export a <code>.txt</code> file with the column names from your reference dataset to later use in the munging of your validation dataset</li>
</ol>
<p>Using GenoML for both your reference dataset and then your validation dataset, the process will look like the following:</p>
<ol>
<li>Munge and train your first dataset
- That will be your “reference” model</li>
<li>Use the outputs of step 1's munge for your reference model to harmonize your incoming validation dataset</li>
<li>Run through harmonization step with your validation dataset</li>
<li>Run through munging with your newly harmonized dataset</li>
<li>Retrain your reference model with only the matching columns of your unseen data
<ul>
<li>Given the nature of ML algorithms, you cannot test a model on a set of data that does not have identical features</li>
</ul></li>
<li>Test your newly retrained reference model on the unseen data</li>
</ol>
<h3><a class="anchor" aria-hidden="true" id="harmonizing-your-validation-test-dataset"></a><a href="#harmonizing-your-validation-test-dataset" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Harmonizing your Validation/Test Dataset</h3>
<p><strong>Required</strong> arguments for harmonizing with GenoML are the following:</p>
<ul>
<li><code>--test_geno_prefix</code> : What is the prefix of your validation dataset PLINK binaries?</li>
<li><code>--test_prefix</code>: What do you want the output to be named?</li>
<li><code>--ref_model_prefix</code>:  What is the name of the previously GenoML-munged dataset you would like to use as your reference dataset? (Without the <code>.dataForML.h5</code> suffix)</li>
<li><code>--training_snps_alleles</code> : What are the SNPs and alleles you would like to use? (This is generated at the end of your previously-GenoML munged dataset with the suffix <code>variants_and_alleles.tab</code>)</li>
</ul>
<p>To harmonize your incoming validation dataset to match the SNPs and alleles to your reference dataset, the command would look like the following:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML harmonize</span>

genoml harmonize \
--test_geno_prefix examples/discrete/validation \
--test_prefix outputs/validation_test_discrete_geno \
--ref_model_prefix outputs/test_discrete_geno \
--training_snps_alleles outputs/test_discrete_geno.variants_and_alleles.tab
</code></pre>
<p>This step will generate:</p>
<ul>
<li>a <code>*_refColsHarmonize_toKeep.txt</code> file of columns to keep for the next step</li>
<li><code>*_refSNPs_andAlleles.*</code> PLINK binary files (.bed, .bim, and .fam) that have the SNPs and alleles match your reference dataset</li>
</ul>
<p>Now that you have harmonized your validation dataset to your reference dataset, you can now munge using a command similar to the following:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> Running GenoML munge after GenoML harmonize</span>

genoml discrete supervised munge --prefix outputs/validation_test_discrete_geno \
--geno outputs/validation_test_discrete_geno_refSNPs_andAlleles \
--pheno examples/discrete/validation_pheno.csv \
--addit examples/discrete/validation_addit.csv \
--ref_cols_harmonize outputs/validation_test_discrete_geno_refColsHarmonize_toKeep.txt
</code></pre>
<p>All munging options discussed above are available at this step, the only difference here is you will add the <code>--ref_cols_harmonize</code> flag to include the <code>*_refColsHarmonize_toKeep.txt</code> file generated at the end of harmonizing to only keep the same columns that the reference dataset had.</p>
<p>After munging and training your reference model and harmonizing and munging your unseen test data, <strong>you will retrain your reference model to include only matching features</strong>. Given the nature of ML algorithms, you cannot test a model on a set of data that does not have identical features.</p>
<p>To retrain your model appropriately, after munging your test data with the <code>--ref_cols_harmonize</code> flag, a final columns list will be generated at <code>*_finalHarmonizedCols_toKeep.txt</code>. This includes all the features that match between your unseen test data and your reference model. Use the <code>--matching_columns</code> flag when retraining your reference model to use the appropriate features.</p>
<p>When retraining of the reference model is complete, you are ready to test!</p>
<p>A step-by-step guide on how to achieve this is listed below:</p>
<pre><code class="hljs css language-shell"><span class="hljs-meta">#</span><span class="bash"> 0. MUNGE THE REFERENCE DATASET</span>
genoml discrete supervised munge \
--prefix outputs/test_discrete_geno \
--geno examples/discrete/training \
--pheno examples/discrete/training_pheno.csv
<span class="hljs-meta">
#</span><span class="bash"> 1. TRAIN THE REFERENCE DATASET</span>
genoml discrete supervised train \
--prefix outputs/test_discrete_geno
<span class="hljs-meta">
#</span><span class="bash"> 2. HARMONIZE TEST DATASET</span>
genoml harmonize \
--test_geno_prefix examples/discrete/validation \
--test_prefix outputs/validation_test_discrete_geno \
--ref_model_prefix outputs/test_discrete_geno \
--training_snps_alleles outputs/test_discrete_geno.variants_and_alleles.tab
<span class="hljs-meta">
#</span><span class="bash"> 3. MUNGE THE TEST DATASET ON REFERENCE MODEL COLUMNS</span>
genoml discrete supervised munge \
--prefix outputs/validation_test_discrete_geno \
--geno outputs/validation_test_discrete_geno_refSNPs_andAlleles \
--pheno examples/discrete/validation_pheno.csv \
--addit examples/discrete/validation_addit.csv \
--ref_cols_harmonize outputs/validation_test_discrete_geno_refColsHarmonize_toKeep.txt
<span class="hljs-meta">
#</span><span class="bash"> 4. RETRAIN REFERENCE MODEL ON INTERSECTING COLUMNS BETWEEN REFERENCE AND TEST</span>
genoml discrete supervised train \
--prefix outputs/test_discrete_geno \
--matching_columns outputs/validation_test_discrete_geno_finalHarmonizedCols_toKeep.txt
<span class="hljs-meta">
#</span><span class="bash"> 5. TEST RETRAINED REFERENCE MODEL ON UNSEEN DATA</span>
genoml discrete supervised test \
--prefix outputs/validation_test_discrete_geno \
--test_prefix outputs/validation_test_discrete_geno \
--ref_model_prefix outputs/test_discrete_geno.trainedModel
</code></pre>
<p><a id="5"></a></p>
<h2><a class="anchor" aria-hidden="true" id="5-experimental-features"></a><a href="#5-experimental-features" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>5. Experimental Features</h2>
<p><strong>UNDER ACTIVE DEVELOPMENT</strong></p>
<p>Planned experimental features include, but are not limited to:</p>
<ul>
<li>Unsupervised munging, training, tuning, and testing</li>
<li>GWAS QC and Pipeline</li>
<li>Network analyses</li>
<li>Meta-learning</li>
<li>Federated learning</li>
<li>Biobank-scale support</li>
<li>Cross-silo checks for genetic duplicates</li>
<li>Outlier detection</li>
<li>...?</li>
</ul>
</span></div></article></div><div class="docs-prevnext"><a class="docs-prev button" href="/docs/contribute"><span class="arrow-prev">← </span><span>Contributing</span></a><a class="docs-next button" href="/docs/usage"><span>Usage</span><span class="arrow-next"> →</span></a></div></div></div><nav class="onPageNav"><ul class="toc-headings"><li><a href="#0-optional-how-to-set-up-a-virtual-environment-via-conda">0. [OPTIONAL] How to Set Up a Virtual Environment via Conda</a></li><li><a href="#1-munging-with-genoml">1. Munging with GenoML</a></li><li><a href="#2-training-with-genoml">2. Training with GenoML</a></li><li><a href="#3-tuning-with-genoml">3. Tuning with GenoML</a></li><li><a href="#4-testing-validation-with-genoml">4. Testing/Validation with GenoML</a><ul class="toc-headings"><li><a href="#harmonizing-your-validation-test-dataset">Harmonizing your Validation/Test Dataset</a></li></ul></li><li><a href="#5-experimental-features">5. Experimental Features</a></li></ul></nav></div><footer class="nav-footer" id="footer"><section class="sitemap"><a href="/" class="nav-home"><img src="/img/dna.png" alt="GenoML" width="66" height="58"/></a><div><h5>Docs</h5><a href="/docs/quick-start">Getting Started</a><a href="/docs/contribute">Contributing</a></div><div><h5>Community</h5><a href="/en/users.html">Users</a></div><div><h5>More</h5><a href="https://github.com/GenoML/">GitHub</a><a class="github-button" href="https://github.com/GenoML/genoml" data-icon="octicon-star" data-count-href="/GenoML/genoml/stargazers" data-show-count="true" data-count-aria-label="# stargazers on GitHub" aria-label="Star this project on GitHub">Star</a></div></section><section class="copyright">Copyright © 2020 GenoML</section></footer></div></body></html>