<!DOCTYPE html><html lang="en"><head><meta charSet="utf-8"/><meta http-equiv="X-UA-Compatible" content="IE=edge"/><title>Usage · GenoML</title><meta name="viewport" content="width=device-width"/><meta name="generator" content="Docusaurus"/><meta name="description" content="&lt;h2&gt;&lt;a class=&quot;anchor&quot; aria-hidden=&quot;true&quot; id=&quot;input-data&quot;&gt;&lt;/a&gt;&lt;a href=&quot;#input-data&quot; aria-hidden=&quot;true&quot; class=&quot;hash-link&quot;&gt;&lt;svg class=&quot;hash-link-icon&quot; aria-hidden=&quot;true&quot; height=&quot;16&quot; version=&quot;1.1&quot; viewBox=&quot;0 0 16 16&quot; width=&quot;16&quot;&gt;&lt;path fill-rule=&quot;evenodd&quot; d=&quot;M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z&quot;&gt;&lt;/path&gt;&lt;/svg&gt;&lt;/a&gt;Input data&lt;/h2&gt;
"/><meta name="docsearch:language" content="en"/><meta property="og:title" content="Usage · GenoML"/><meta property="og:type" content="website"/><meta property="og:url" content="https://genoml.github.io/index.html"/><meta property="og:description" content="&lt;h2&gt;&lt;a class=&quot;anchor&quot; aria-hidden=&quot;true&quot; id=&quot;input-data&quot;&gt;&lt;/a&gt;&lt;a href=&quot;#input-data&quot; aria-hidden=&quot;true&quot; class=&quot;hash-link&quot;&gt;&lt;svg class=&quot;hash-link-icon&quot; aria-hidden=&quot;true&quot; height=&quot;16&quot; version=&quot;1.1&quot; viewBox=&quot;0 0 16 16&quot; width=&quot;16&quot;&gt;&lt;path fill-rule=&quot;evenodd&quot; d=&quot;M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z&quot;&gt;&lt;/path&gt;&lt;/svg&gt;&lt;/a&gt;Input data&lt;/h2&gt;
"/><meta property="og:image" content="https://genoml.github.io/img/docusaurus.png"/><meta name="twitter:card" content="summary"/><meta name="twitter:image" content="https://genoml.github.io/img/docusaurus.png"/><link rel="shortcut icon" href="/img/favicon.png"/><link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css"/><script>
              (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
              (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
              m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
              })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

              ga('create', 'UA-134657372-1', 'auto');
              ga('send', 'pageview');
            </script><script type="text/javascript" src="https://buttons.github.io/buttons.js"></script><link rel="stylesheet" href="/css/main.css"/><script src="/js/codetabs.js"></script></head><body class="sideNavVisible separateOnPageNav"><div class="fixedHeaderContainer"><div class="headerWrapper wrapper"><header><a href="/"><img class="logo" src="/img/dna.png" alt="GenoML"/><h2 class="headerTitleWithLogo">GenoML</h2></a><div class="navigationWrapper navigationSlider"><nav class="slidingNav"><ul class="nav-site nav-site-internal"><li class="siteNavGroupActive"><a href="/docs/about" target="_self">About</a></li><li class="siteNavGroupActive"><a href="/docs/quick-start" target="_self">Getting Started</a></li><li class="siteNavGroupActive"><a href="/docs/help" target="_self">Help</a></li><li class=""><a href="https://github.com/GenoML" target="_self">GitHub</a></li></ul></nav></div></header></div></div><div class="navPusher"><div class="docMainWrapper wrapper"><div class="container docsNavContainer" id="docsNav"><nav class="toc"><div class="toggleNav"><section class="navWrapper wrapper"><div class="navBreadcrumb wrapper"><div class="navToggle" id="navToggler"><i></i></div><h2><i>›</i><span>Getting Started</span></h2><div class="tocToggler" id="tocToggler"><i class="icon-toc"></i></div></div><div class="navGroups"><div class="navGroup"><h3 class="navGroupCategoryTitle">Intro</h3><ul class=""><li class="navListItem"><a class="navItem" href="/docs/about">About</a></li><li class="navListItem"><a class="navItem" href="/docs/help">Help</a></li><li class="navListItem"><a class="navItem" href="/docs/roadmap">Roadmap</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">Getting Started</h3><ul class=""><li class="navListItem"><a class="navItem" href="/docs/quick-start">Quick start</a></li><li class="navListItem navListItemActive"><a class="navItem" href="/docs/usage">Usage</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">How it works</h3><ul class=""><li class="navListItem"><a class="navItem" href="/docs/workflow">Workflow</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">API Reference</h3><ul class=""><li class="navListItem"><a class="navItem" href="/docs/cli">The Command Line Interface (CLI)</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">Contributing </h3><ul class=""><li class="navListItem"><a class="navItem" href="/docs/contribute">Developing GenoML</a></li></ul></div></div></section></div><script>
            var coll = document.getElementsByClassName('collapsible');
            var checkActiveCategory = true;
            for (var i = 0; i < coll.length; i++) {
              var links = coll[i].nextElementSibling.getElementsByTagName('*');
              if (checkActiveCategory){
                for (var j = 0; j < links.length; j++) {
                  if (links[j].classList.contains('navListItemActive')){
                    coll[i].nextElementSibling.classList.toggle('hide');
                    coll[i].childNodes[1].classList.toggle('rotate');
                    checkActiveCategory = false;
                    break;
                  }
                }
              }

              coll[i].addEventListener('click', function() {
                var arrow = this.childNodes[1];
                arrow.classList.toggle('rotate');
                var content = this.nextElementSibling;
                content.classList.toggle('hide');
              });
            }

            document.addEventListener('DOMContentLoaded', function() {
              createToggler('#navToggler', '#docsNav', 'docsSliderActive');
              createToggler('#tocToggler', 'body', 'tocActive');

              const headings = document.querySelector('.toc-headings');
              headings && headings.addEventListener('click', function(event) {
                if (event.target.tagName === 'A') {
                  document.body.classList.remove('tocActive');
                }
              }, false);

              function createToggler(togglerSelector, targetSelector, className) {
                var toggler = document.querySelector(togglerSelector);
                var target = document.querySelector(targetSelector);

                if (!toggler) {
                  return;
                }

                toggler.onclick = function(event) {
                  event.preventDefault();

                  target.classList.toggle(className);
                };
              }
            });
        </script></nav></div><div class="container mainContainer"><div class="wrapper"><div class="post"><header class="postHeader"><h1 class="postHeaderTitle">Usage</h1></header><article><div><span><h2><a class="anchor" aria-hidden="true" id="input-data"></a><a href="#input-data" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Input data</h2>
<p>Here is a quick walk through of input file formats and general suggestions.</p>
<blockquote>
<p>Example data: You can use the IPDGC (International Parkinson's Disease Genomics Consortium) example data, available <a href="https://github.com/ipdgc/GenoML-Brief-Intro/raw/master/exampleData.zip">here</a>.</p>
</blockquote>
<h3><a class="anchor" aria-hidden="true" id="genotypes-required"></a><a href="#genotypes-required" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Genotypes (required)</h3>
<p>Generally these should be single nucleotide polymorphisms and samples passing standard GWAS type quality controls.
If using imputed data, we suggest filtering at an imputation quality &gt; 0.8 and using the hard call genotypes.
Additionally, rarer variants can be problematic, and we suggest filtering at a minor allele frequency &gt; 1% (or &gt; 5%) if sample size permits. The fewer samples you have, the less you should focus on variants that have lower minor allele frequencies.
Limiting missingness in these files is helpful even though there is secondary imputation as part of GenoML.
This input data is the standard .bed, .bim and .fam binaries from <a href="https://www.cog-genomics.org/plink/1.9/input#bed">PLINK</a>.</p>
<h3><a class="anchor" aria-hidden="true" id="phenotypes-required"></a><a href="#phenotypes-required" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Phenotypes (required)</h3>
<p>A basic three column tab delimited file as per PLINK specifications with the column headers FID, IID and PHENO corresponding to samples in the genotype data.<br>
In general, there should be no missing data.  Please code discrete phenotypes as 1/2, with 2 as a case and 1 as a control. Continuous phenotypes should be relatively normally distributed if possible.
File suffix should be &quot;.pheno&quot;.</p>
<h3><a class="anchor" aria-hidden="true" id="covariates-optional"></a><a href="#covariates-optional" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Covariates (optional)</h3>
<p>Tab delimited file with the first header columns FID and IID, the following columns can contain any numeric data deemed necessary.<br>
In general we use this for principal components to adjust for population substructure at the variant selection phase.
File suffix should be &quot;.cov&quot;.</p>
<h3><a class="anchor" aria-hidden="true" id="additional-optional"></a><a href="#additional-optional" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Additional (optional)</h3>
<p>Tab delimited file with the first header columns FID and IID, the following columns can contain any numeric data deemed necessary.<br>
In general we use this for parameters we want to use as predictors that aren't SNPs.  For example, this could include clinical data or thousands of gene expression probes, whatever you want.
File suffix should be &quot;.addit&quot;.</p>
<h3><a class="anchor" aria-hidden="true" id="gwas-optional"></a><a href="#gwas-optional" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>GWAS (optional)</h3>
<p>This is a big tab delimited text file of genome-wide association study summary stats.
This file must have header as follows, <em>SNP A1 A2 freq b se p N</em>, where <em>SNP</em> is a unique variant ID, <em>A1</em> is the effect allele, <em>A2</em> is the reference allele, <em>freq</em> is the frequency of A1, <em>b</em> is the beta coefficient from GWAS, <em>se</em> is the standard error from GWAS, <em>p</em> is the p-value from GWAS and <em>N</em> is the sample size. No missing data is allowed.</p>
<h2><a class="anchor" aria-hidden="true" id="train-the-ml-model"></a><a href="#train-the-ml-model" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Train the ML model</h2>
<p>This step performs data pruning as well as model training and tunning.</p>
<h3><a class="anchor" aria-hidden="true" id="basic"></a><a href="#basic" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Basic</h3>
<p>Requires <code>genotype</code> and <code>phenotype</code> data, run:</p>
<pre><code class="hljs">genoml-train <span class="hljs-params">--geno-prefix=</span><span class="hljs-string">./exampleData/training</span> <span class="hljs-params">--pheno-file=</span><span class="hljs-string">./exampleData/training.pheno</span> <span class="hljs-params">--model-file=</span><span class="hljs-string">./exampleModel</span>
</code></pre>
<h3><a class="anchor" aria-hidden="true" id="with-external-gwas-summary-stats"></a><a href="#with-external-gwas-summary-stats" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>With external GWAS summary stats</h3>
<p>Using <code>genotype</code>, <code>phenotype</code> , and <code>GWAS</code> data, run:</p>
<pre><code class="hljs">genoml-train <span class="hljs-params">--geno-prefix=</span><span class="hljs-string">./exampleData/training</span> <span class="hljs-params">--pheno-file=</span><span class="hljs-string">./exampleData/training.pheno</span>  <span class="hljs-params">--gwas-file=</span><span class="hljs-string">./exampleData/example_GWAS.txt</span> <span class="hljs-params">--model-file=</span><span class="hljs-string">./exampleModel</span>
</code></pre>
<h3><a class="anchor" aria-hidden="true" id="adding-convariates-and-additional-predictor-files"></a><a href="#adding-convariates-and-additional-predictor-files" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Adding convariates and additional predictor files</h3>
<p>Using <code>genotype</code>, <code>phenotype</code> , <code>GWAS</code>, <code>covariate</code>, and <code>additional</code> data, run:</p>
<pre><code class="hljs">genoml-train <span class="hljs-params">--geno-prefix=</span><span class="hljs-string">./exampleData/training</span> <span class="hljs-params">--pheno-file=</span><span class="hljs-string">./exampleData/training.pheno</span> <span class="hljs-params">--gwas-file=</span><span class="hljs-string">./exampleData/example_GWAS.txt</span> <span class="hljs-params">--cov-file=</span><span class="hljs-string">./exampleData/training.cov</span> <span class="hljs-params">--addit-file=</span><span class="hljs-string">./exampleData/training.addit</span> <span class="hljs-params">--model-file=</span><span class="hljs-string">./exampleModel</span> 
</code></pre>
<h3><a class="anchor" aria-hidden="true" id="optional-flags"></a><a href="#optional-flags" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Optional flags</h3>
<p>The following flags provide more flexibility:</p>
<table>
<thead>
<tr><th>Flag</th><th>Description</th></tr>
</thead>
<tbody>
<tr><td>--no-tune</td><td>Disable Tuning. This marginally reduces the accuracy of the trained model but significately increases the training time.</td></tr>
<tr><td>--n-cores=</td><td>Number of cores to be allocated for computation [default: 1].</td></tr>
<tr><td>--train-speed=</td><td>Training speed: (ALL, FAST, FURIOUS, BOOSTED). Run all models, only,the fastest models, run slightly slower models, or just run boosted models which usually perform best when using genotype data [default: BOOSTED].</td></tr>
<tr><td>--cv-reps=</td><td>Number of cross-validation. An integer greater than 5. Effects the speed [default: 5].</td></tr>
<tr><td>--impute-data=</td><td>Imputation: (knn, median). Governs secondary imputation and data transformation [default: median].</td></tr>
<tr><td>--grid-search=</td><td>Grid search length for parameters, integer greater than 10, 30 or greater recommended, effects speed of initial tune [default: 10].</td></tr>
</tbody>
</table>
<h2><a class="anchor" aria-hidden="true" id="using-the-trained-ml-model-for-inference"></a><a href="#using-the-trained-ml-model-for-inference" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Using the trained ML model for inference</h2>
<p>To perform inference or external validation only when <code>genotype</code> and <code>phenotype</code> data present, run:</p>
<pre><code class="hljs">genoml-inference <span class="hljs-params">--model-file=</span><span class="hljs-string">./exampleModel</span> <span class="hljs-params">--valid-geno-prefix=</span><span class="hljs-string">./exampleData/validation</span> <span class="hljs-params">--valid-pheno-file=</span><span class="hljs-string">./exampleData/validation.pheno</span>
</code></pre>
</span></div></article></div><div class="docs-prevnext"><a class="docs-prev button" href="/docs/quick-start"><span class="arrow-prev">← </span><span>Quick start</span></a><a class="docs-next button" href="/docs/workflow"><span>Workflow</span><span class="arrow-next"> →</span></a></div></div></div><nav class="onPageNav"><ul class="toc-headings"><li><a href="#input-data">Input data</a><ul class="toc-headings"><li><a href="#genotypes-required">Genotypes (required)</a></li><li><a href="#phenotypes-required">Phenotypes (required)</a></li><li><a href="#covariates-optional">Covariates (optional)</a></li><li><a href="#additional-optional">Additional (optional)</a></li><li><a href="#gwas-optional">GWAS (optional)</a></li></ul></li><li><a href="#train-the-ml-model">Train the ML model</a><ul class="toc-headings"><li><a href="#basic">Basic</a></li><li><a href="#with-external-gwas-summary-stats">With external GWAS summary stats</a></li><li><a href="#adding-convariates-and-additional-predictor-files">Adding convariates and additional predictor files</a></li><li><a href="#optional-flags">Optional flags</a></li></ul></li><li><a href="#using-the-trained-ml-model-for-inference">Using the trained ML model for inference</a></li></ul></nav></div><footer class="nav-footer" id="footer"><section class="sitemap"><a href="/" class="nav-home"><img src="/img/dna.png" alt="GenoML" width="66" height="58"/></a><div><h5>Docs</h5><a href="/docs/quick-start">Getting Started</a><a href="/docs/contribute">Contributing</a></div><div><h5>Community</h5><a href="/en/users.html">Users</a></div><div><h5>More</h5><a href="https://github.com/GenoML/">GitHub</a><a class="github-button" href="https://github.com/GenoML/genoml" data-icon="octicon-star" data-count-href="/GenoML/genoml/stargazers" data-show-count="true" data-count-aria-label="# stargazers on GitHub" aria-label="Star this project on GitHub">Star</a></div></section><section class="copyright">Copyright © 2019 GenoML</section></footer></div></body></html>